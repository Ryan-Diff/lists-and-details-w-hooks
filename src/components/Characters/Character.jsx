import React from 'react';
import PropTypes from 'prop-types';

const Character = ({ name, photoUrl, weapon, position, affiliation }) => (
    <figure>
        <h1>{name}</h1>
        <img src={photoUrl} />
        <h3>{weapon}</h3>
        <h3>{position}</h3>
        <h3>{affiliation}</h3>
    </figure>
)

Character.propTypes = {
    name: PropTypes.string.isRequired,
    photoUrl: PropTypes.string.isRequired,
    weapon: PropTypes.string.isRequired,
    position: PropTypes.string.isRequired,
    affiliation: PropTypes.string.isRequired
}

export default Character;