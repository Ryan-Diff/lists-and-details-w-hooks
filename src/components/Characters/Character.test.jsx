import React from 'react';
import { render } from '@testing-library/react';
import Character from './Character';

describe('Character component', () => {
    it('renders a character', () => {
        const { asFragment } = render(
            <Character
              name='Ryan'
              photoUrl='Test'
              weapon='Anxiety'
              position='Serf'
              affiliation='Alchemy'
            />);
            expect(asFragment()).toMatchSnapshot();
    });
});