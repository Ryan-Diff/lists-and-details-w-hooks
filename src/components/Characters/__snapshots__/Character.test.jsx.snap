// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders a character 1`] = `
<DocumentFragment>
  <figure
    data-testid="detail"
  >
    <h1>
      Ryan
    </h1>
    <img
      src="Test"
    />
    <h3>
      Anxiety
    </h3>
    <h3>
      Serf
    </h3>
    <h3>
      Alchemy
    </h3>
  </figure>
</DocumentFragment>
`;
